server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -15,39 +17,106 @@
         URI youtubeUri = URI.create("https://www.googleapis.com/youtube/v3/videos?" + youtubeParam.build());
 
         var httpClient = HttpClient.newHttpClient();
-        var youtubeReq = HttpRequest.newBuilder(youtubeUri)
-                .header("Authorization", "Bearer " + accessToken.toString())
+        HttpRequest youtubeReq = HttpRequest.newBuilder(youtubeUri)
+                .header("Authorization", "Bearer " + accessToken)
                 .header("Accept", "application/json")
                 .GET()
                 .build();
 
-        // get response to Youtube API 
+        // get response from Youtube API 
         var youtubeRes = httpClient.sendAsync(youtubeReq, BodyHandlers.ofString()).join();
-        var youtubeResBody = youtubeRes.body();
-        res.getWriter().write(youtubeResBody);
+        return youtubeRes.body();
+    }
+
+    /**
+     * checks whether topic is categorized as music
+     * by checking if the last word is "music" or "Music"
+     * @param topic identifies youtube video category e.g. Knowledge or Pop music
+     * @return whether topic is categorized as music
+     */
+    protected Boolean isMusic(String topic) {
+        String lastWord = topic.substring(topic.lastIndexOf(" ") + 1);
+        return lastWord.equalsIgnoreCase("music");
+    }
+
+    /**
+     * parses through youtube liked videos json string,
+     * updates hash map to contain frequency count of each music genre
+     * @param youtubeResBody json response of youtube liked videos
+     * @param genreCount hash map of frequency count of each music genre
+     */
+    protected void updateMusicCount(String youtubeResBody, HashMap<String, Integer> genreCount) {
+        JsonObject jObject = JsonParser.parseString(youtubeResBody).getAsJsonObject();
+        JsonArray videos = jObject.getAsJsonArray("items");
+
+        // iterates through each liked video
+        for (int i = 0; i < videos.size(); i++) {
+            // extracts array of topicCategories in video
+            JsonObject video = videos.get(i).getAsJsonObject();
+            JsonObject topicDetails = video.getAsJsonObject("topicDetails");
+            if (topicDetails == null) {
+                // skip this video if it doesn't have topic details
+                continue;
+            }
+            JsonArray topicCategories = topicDetails.getAsJsonArray("topicCategories");
+
+            for (int j = 0; j < topicCategories.size(); j++) {
+                // extract music genre out of wikipedia links of topic categories
+                String link = topicCategories.get(j).toString();
+                String topic = link.substring(link.lastIndexOf('/') + 1);
+                topic = topic.replaceAll("\"", "");
+                topic = topic.replaceAll("_", " ");
+
+                // skip if topic is not music
+                if (!isMusic(topic)) {
+                    break;
+                }
+
+                // update genreCount hashmap
```

Comment:
Same here, saying that you're adding something to a hashmap doesn't tell me anything about the intent of this code, just the mechanics